siikr/internal/globals.php

```diff
@@ -368,3 +368,10 @@ function check_delete($tag_text, $prior_count, $blog_uuid, $db) {
     }
     return false;
 }
+
+/**
+ * Get a handle to the database.
+ */
+function get_db() {
```

Comment:
The db may be signed into by two separate users. One as www-data for client accessible scripts, and one as a system user for long running tasks that don't eat up a server thread. So this abstraction would ideally accept an argument to differentiate.

Whichever the approach, most of the variables required are imported from the `auth/credentials.php` file, which means we have to specify `global` in order to access them.

```php
function get_db() {
    global $db_host; 
    global $db_name;
    global $db_user;
    global $db_pass;

    return new PDO("pgsql:host=$db_host;dbname=$db_name", $db_user, $db_pass);
}
```

I will be making these globals into proper constants with the next push. But for now I suspect this is why indexing isn't working for you.

Reply:
OK.

Reply:
So undo that use of globals and reset all the get_db() calls #5 built?

Reply:
Yeah, just for now. If for no other reason than to minimize implementation confounders.

Reply:
So we'll probably need to setup the www-data user in the sql somehow since these are two isolated users here.

Reply:
I *thought* I had done that already as part of the schema and setup_simple.sh. But apparently not. It should be to 
the script's php_user, and not the harcoded www-data though. 

Man I really should have spent more time cleaning this up before publishing it.

Reply:
Take a look at my latest push?

Reply:
Anyway, siikrweb should be fine for all of the basic functionality tasks I think. You're unlikely to need any of the maintenance functions for a little while.